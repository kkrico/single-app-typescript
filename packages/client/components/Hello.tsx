import * as React from 'react';

export interface HelloProps {
    compiler: string;
    framework: string;
}

// 'HelloProps' describes the shape of props.
// State is never set so we use the '{}' type.
export const HelloWorld: React.FC<HelloProps> = ({ compiler, framework }: HelloProps) => {
    const [message, setMessage] = React.useState(null);

    React.useEffect(() => {
        const fetchItens = async (): Promise<void> => {
            const apiUrl = `${process.env.API_URL}:${process.env.API_PORT}${process.env.API_VERSION}`;
            const result = await fetch(`${apiUrl}/hello-world`);
            const msg = await result.json();

            setMessage(msg);
        };

        fetchItens();
    }, []);

    return (
        <>
            <h1>
                Hello from {compiler} and {framework}
            </h1>
            {message != null && <p>{JSON.stringify(message)}</p>}
        </>
    );
};
