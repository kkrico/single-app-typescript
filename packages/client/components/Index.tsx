import { HelloWorld } from './Hello';
import * as React from 'react';

const App: React.FC<{}> = () => {
    return <HelloWorld compiler="Typescript" framework="React"></HelloWorld>;
};

export default App;
