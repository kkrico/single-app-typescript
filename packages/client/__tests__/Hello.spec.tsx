import * as React from 'react';
import { shallow } from 'enzyme';
import { HelloWorld } from '../components/Hello';

describe('Hello World', () => {
    it('Render sem crash', () => {
        const link = shallow(<HelloWorld compiler="Typescript" framework="React" />);
        expect(link).toMatchSnapshot();
    });
});
