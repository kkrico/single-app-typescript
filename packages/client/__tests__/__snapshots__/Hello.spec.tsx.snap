// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World Render sem crash 1`] = `
<Fragment>
  <h1>
    Hello from 
    Typescript
     and 
    React
  </h1>
</Fragment>
`;
